### PR TITLE
Check those headers

### LINKED ISSUE #46: update checks.sh to include header files
As a daft engineer I want to ensure header files match our code guidelines.

Need to update `SRC` assignment to:
```bash
SRC=$(find src \( -name "*.cc" -o -name "*.hh" \))
```

Success criteria:
- checks.sh checks header files
- Any found errors are resolved

### PR DESCRIPTION
**Pull request by Jambls**
Updated checks.sh to make it check headers
Omitted the llvm-header-guard check because making it make sense would have been a lot of effort.

Closes #46

### CODE REVIEW

**Review by MHCooke — DISMISSED** [review]

File: checks.sh [hunk 1]
```diff
@@ -2,21 +2,22 @@
 
 set -e
 
-SRC=$(find src -name "*.cc")
+INC=$(find include -name "*.hh")
+SRC=$(find src \( -name "*.cc" -o -name "*.hh" \))
```
Comment (MHCooke) [inline checks.sh:6]: I think we should probably raise an error if header files are found in src?

**Review by MHCooke — CHANGES_REQUESTED** [review]
.clang-tidy has been emptied

**Comment by MHCooke** [comment]
Do you have a separate patch for #55 , or is there is a missing commit here

**Comment by Jambls** [comment]
> .clang-tidy has been emptied

It hasn't, I think github is bugging out because the file has been changed twice, but over the whole PR it's back to where it was.

**Comment by Jambls** [comment]
> Do you have a separate patch for #55 , or is there is a missing commit here

I was gonna do a separate PR, it's a separate issue and separate problem. Happy to combine them if you want.

File: checks.sh (hunk 1 above)
Comment (Jambls) [reply checks.sh:6]: I'll look into it

**Comment by Jambls** [comment]
Now that I've moved the header files, I should check cmake to find out if they'll still be found by it

File: checks.sh (hunk 1 above)
Comment (Jambls) [reply checks.sh]: Fixed

**Comment by MarkAHarley** [comment]
Why does each include file need a subdir now that there is an include folder? i.e it seems unnecessary to have include/filter/filter.hh when you could just have include/filter.hh

**Review by MarkAHarley — APPROVED** [review]
Looks good.

File: checks.sh [hunk 2]
```diff
@@ -2,22 +2,30 @@
...
 else
-    clang-tidy -p build $SRC
-    clang-format -i -style=file $SRC $UNIT
+    clang-tidy -p build $INC $SRC 
+    clang-format -i -style=file $INC $SRC $UNIT
     if [[ -n $WRONG_EXT ]]; then
         echo "Files found ending in .cpp: $WRONG_EXT"
+    elif [[ -n $WRONG_LOCATION ]]; then 
+	echo "Header files found in src or cc files found in include"
+	exit 1
```
Comment (MHCooke) [inline checks.sh:29]: You should make it print out the files found that violate the rule.

**Comment by MHCooke** [comment]
> Why does each include file need a subdir now that there is an include folder? i.e it seems unnecessary to have include/filter/filter.hh when you could just have include/filter.hh

@Jambls do you have any thoughts on this? I was going from my previous (python) experience and putting everything in sub dirs, but if we aren't going to have subclasses of things, it does seem sensible to keep them all in one place.

**Review by MHCooke — COMMENTED** [review]
Update checks,sh to print error locations for files in wrong place.

(mostly posting this to see if it clears the "MHCooke requested changes" flag)

File: checks.sh (hunk 2 above)
Comment (Jambls) [reply checks.sh:29]: done

**Review by MHCooke — APPROVED** [review]

**Pull request merged**